### ISSUE TITLE
ColorPicker crashes application

### ISSUE THREAD

**Issue by xceedsoftware** [xceedsoftware/wpftoolkit#725]
<b>JLT93[CodePlex]</b> <br />Hello, this problem does not happen to me however I let someone else use my application for testing purposes and whenever he opens up a window containing the ColorPicker, the application would stop responding.

I have isolated the issue to be related to the ColorPicker only as when I commented out the XAML for it, it was no longer a problem.

I use the ColorPicker in a dialog window which is shown. Everything works fine without the ColorPicker.

System specs of user:
CPU - Intel Core i3-2310M @ 2.10GHz 
RAM - 2.00 GB
Type - 64-bit
OS - Windows 7 SP1
Running .NET Framework 4.5

My application is compiled with .NET Framework 4.5 however I am using 4.5.1, unlikely to be the problem.

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />I'm not sure I understand. 
Compiling the Toolkit will result in a Xceed.Wpf.Toolkit.dll file. That file should be reference by any external project that want to use a Toolkit's Control. So if his project references the dll, it should be fine. There is no embedding of the Xceed.Wpf.Toolkit.dll.

**Comment by xceedsoftware** [comment]
<b>JLT93[CodePlex]</b> <br />Yes. I was just wondering if you can sort of embed the Xceed.Wpf.Toolkit.dll inside the application itself instead of having to share it. If you can't do that then it's all fine, it's not a problem to supply it.

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

Yes, that can explain the crash. As soon as the application tries to use the ColorPicker from Xceed.Wpf.Toolkit.dll and can't find it, it crashes.


Can you share your Xceed.Wpf.Toolkit.dll with this user ? 

Normally, when compiling the Xceed.Wpf.Toolkit, the Xceed.Wpf.Toolkit.dll should be put in the output folder. If not, there is a problem when compiling, or the output folder is somewhere else.


In conclusion, the Xceed.Wpf.Toolkit.dll is necessary in order to use the ColorPicker.

**Comment by xceedsoftware** [comment]
<b>JLT93[CodePlex]</b> <br />I was looking in my debug folder, and I seen the Xceed.Wpf.Toolkit.dll file. I had a lightbulb shining above my head and now I know what the problem is. I have a copy of the .dll so it works fine, however the user that is having the crashing issue does
 not have the .dll file. That would be exactly why there is a problem. 

Is there a way to compile the app without having the .dll show up in the output folder?

**Comment by xceedsoftware** [comment]
<b>JLT93[CodePlex]</b> <br />Is there any way I could further debug the issue? It may be a hardware issue, either not good enough to handle it or a possibly faulty ram segment but cannot be sure. Surely there must be a way I can detect why an application crashes though?

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hum, 

Sorry, I can not reproduce this issue in Windows7 or Windows8, with .Net 4.5.

**Comment by xceedsoftware** [comment]
<b>JLT93[CodePlex]</b> <br />Well there really isn't that much to it. If I comment out the call to xctk:ColorPicker then the window loads fine for them.

Window x:Class=MyApp.Settings
        xmlns=http://schemas.microsoft.com/winfx/2006/xaml/presentation
        xmlns:x=http://schemas.microsoft.com/winfx/2006/xaml
        xmlns:xctk=http://schemas.xceed.com/wpf/xaml/toolkit
        Title=Settings Height=300 Width=300 ResizeMode=NoResize WindowStartupLocation=CenterOwner WindowStyle=ThreeDBorderWindow
    Grid Margin=5
        DockPanel LastChildFill=True
            Grid DockPanel.Dock=Bottom
                Button Content=Cancel HorizontalAlignment=Left Width=75 IsCancel=True/
                Button Content=Save HorizontalAlignment=Right Width=75 IsDefault=True Click=Button_Click/
            /Grid
            Grid
                GroupBox Header=Settings Margin=0,0,0,5
                    Grid
                        Grid.RowDefinitions
                            RowDefinition Height=Auto /
                            RowDefinition Height=Auto /
                            RowDefinition Height=Auto /
                        /Grid.RowDefinitions
                        Label FontWeight=Bold Grid.Row=0Username color/Label
                        TextBlock Grid.Row=1 TextWrapping=WrapWithOverflowThis will allow you to change what color your username is when displayed to yourself and other users./TextBlock
                        xctk:ColorPicker Grid.Row=2 DisplayColorAndName=True ShowStandardColors=False UsingAlphaChannel=False Name=colorPicker x:FieldModifier=public/xctk:ColorPicker
                    /Grid
                /GroupBox
            /Grid
        /DockPanel
    /Grid
/Window

The C# I use to display the window is the same as follows:
        Settings window = new Settings(this.chatSettings) {
            Owner = this
        };
        Nullablebool dialogResult = window.ShowDialog();

However I wouldn't believe that to be an issue. To me it looks like their system specs cannot handle the size of the colorpicker. Even I have some troubles but only when opening the colorpicker, it will be a little slow for a couple of seconds but then it's
 normal.

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

Can you paste the code that cause the bug : Window and colorPicker ? 

Thank you.
